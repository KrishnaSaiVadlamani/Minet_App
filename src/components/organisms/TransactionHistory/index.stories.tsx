import { ComponentStory, ComponentMeta } from '@storybook/react'
import TransactionHistory from './index'

export default {
  title: 'Organisms/TransactionHistory',
  component: TransactionHistory,
} as ComponentMeta<typeof TransactionHistory>

const Template: ComponentStory<typeof TransactionHistory> = () => (
  <TransactionHistory/>
)

export const SimpleTransactionHistory = Template.bind({})