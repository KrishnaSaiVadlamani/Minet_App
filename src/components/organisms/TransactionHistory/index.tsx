import { Grid, styled, Stack, Box } from '@mui/material'
import { BitcoinTransactionValues } from '../../../utils/constants'
import Success from '../../../../public/assets/images/success.svg'
import Failure from '../../../../public/assets/images/failed.svg'
import Pending from '../../../../public/assets/images/processed.svg'
import TypographyComponent from '../../atoms/Typography'
import theme from '../../../theme'
import IconWithTypography from '../../molecules/IconWithTypography'
import ChipItem from '../../atoms/ChipItem'
import ValueChangeTypographyComponent from '../../molecules/ValueChangeTypography'
import Slider from '../../../../public/assets/images/slider.svg'
import IconComponent from '../../atoms/Icons'
import React from 'react'
import { Transaction } from '../../../utils/types'

interface CoinProps {
  name: string
  from: string
  date: number
  month: string
  btcPrice: number
  dollarPrice: number
  status: string

}
interface TransactionHistoryProps{
  transactions: Transaction[]
}

const StyledContainer = styled(Grid)({
  width: '100%',
  height: '100%',
  border: `1px solid ${theme.palette.greyColors.grey100}`,
  borderRadius: '0px 0px 4px 4px',
  margin: '0px !important',
  boxSizing: 'border-box',
})

const StyledGridContainer = styled(Grid)({
  width: '100%',
  borderBottom: `1px solid ${theme.palette.greyColors.grey100}`,
  '&:last-child': {
    borderBottom: 'none',
  },
  display: 'flex',
})

const LeftContainer = styled(Grid)({
  padding: '12px 24px 0px 24px',
})

const RightContainer = styled(Grid)({
  paddingTop: '25px',
  display: 'flex',
  justifyContent: 'center',
})

const ChipBox = styled(Box)({
  paddingTop: '23px',
})

const renderDate = (date: number, month: string) => {
  return (
    <Box>
      <TypographyComponent
        variant="caption2"
        children={month}
        color={theme.palette.textColor.mediumEmphasis}
      />
      <TypographyComponent
        variant="subtitle2"
        children={date.toString()}
        color={theme.palette.textColor.highEmphasis}
      />
    </Box>
  )
}

const ImageType = (status: string) => {
  if (status === 'success') {
    return Success
  } else if (status === 'pending') {
    return Pending
  }

  return Failure
}

const renderTypography = (status: string, name: string, from: string) => {
  return (
    <IconWithTypography
      image={ImageType(status)}
      imageHeight={'44px'}
      imageWidth={'44px'}
      text={name}
      textVariant={'body1'}
      textColor={theme.palette.textColor.highEmphasis}
      subText={`From ${from}`}
      subTextVariant={'caption2'}
      subTextColor={theme.palette.textColor.mediumEmphasis}
    />
  )
}

const renderChipItem = () => {
  return (
    <ChipBox>
      <ChipItem label={'Purchased'} chiptype={'rounded'} chipvariant={'dark'} />
    </ChipBox>
  )
}

const renderLeftComponent = (coin: Transaction) => {
  return (
    <Stack direction="row" alignItems={'center'}>
      {renderDate(coin.date, coin.month)}
      {renderTypography(coin.status, coin.name, coin.from)}
      {renderChipItem()}
    </Stack>
  )
}

const renderCards = (transactions: Transaction[]) => {
  console.log("Transactins ",transactions)
  return transactions.map((coin, index) => {
    return (
      <StyledGridContainer container direction={'row'} alignItems={'center'}>
        <Grid item key={index} xs={11}>
          {renderLeftComponent(coin)}
        </Grid>
        <Grid item alignItems={'flex-end'} xs={1}>
          <ValueChangeTypographyComponent
            topText={`+${coin.quantity} ${coin.symbol}`}
            bottomText={`+$${coin.price}`}
            topTextColor={theme.palette.textColor.highEmphasis}
            bottomTextColor={theme.palette.textColor.mediumEmphasis}
          />
        </Grid>
      </StyledGridContainer>
    )
  })
}

const TransactionHistory = (props:TransactionHistoryProps) => {
  return (
    <StyledContainer container data-testid="transaction-history">
      <LeftContainer item xs={11.8}>
        {renderCards(props.transactions)}
      </LeftContainer>
      <RightContainer item xs={0.2}>
        <IconComponent src={Slider}  />
      </RightContainer>
    </StyledContainer>
  )
}

export default TransactionHistory
