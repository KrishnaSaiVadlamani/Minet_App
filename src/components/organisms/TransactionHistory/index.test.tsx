import renderer from 'react-test-renderer';
import TransactionHistory from '.';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';
it('renders transaction history correctly', () => {
    const tree = renderer
      .create( <TransactionHistory transactions={[]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Organisms/TransactionHistory', () => {
    test('Renders transaction history correctly', () => {
        render( <TransactionHistory transactions={[]} />);
        expect(screen.getByTestId('transaction-history')).toBeInTheDocument();
    })
})