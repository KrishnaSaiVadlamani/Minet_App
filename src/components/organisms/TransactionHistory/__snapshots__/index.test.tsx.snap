// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders transaction history correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-5jjnw1-MuiGrid-root"
  data-testid="transaction-history"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11.8 css-zbbsip-MuiGrid-root"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-0.2 css-z96tjo-MuiGrid-root"
  >
    <div>
      <img
        alt="icon"
        className="css-0"
        data-testid="iconComponent"
        src={
          {
            "process": [Function],
          }
        }
      />
    </div>
  </div>
</div>
`;
